app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_history = {}
+
+    def wrapper(*args):
+        nonlocal cache_history
+
+        if args in cache_history.keys():
+            print("Getting from cache")
+            return cache_history[args]
+        else:
+            print("Calculating new result")
+            result = func(args)
```

Comment:
The function call `func(args)` should be `func(*args)` to correctly pass the arguments to the decorated function. This is crucial for the decorator to work with functions that have varying numbers of arguments.